File: c++/Komponen.cpp
#include <iostream>
using namespace std;
#include "Elektronik.cpp"

class Komponen : public Elektronik {
protected:
    string kode;
    string tipe;

public:
    Komponen(string merk, int tahunProduksi, double harga, string kode, string tipe);
    string getKode();
    void setKode(string kode);
    string getTipe();
    void setTipe(string tipe);
    void displayInfo() override;
};

string Komponen::getKode() { return kode; }
void Komponen::setKode(string kode) { this->kode = kode; }

string Komponen::getTipe() { return tipe; }
void Komponen::setTipe(string tipe) { this->tipe = tipe; }

void Komponen::displayInfo() {
    Elektronik::displayInfo();
    cout << "Kode: " << kode << endl;
    cout << "Tipe: " << tipe << endl;
}

File: java/Komponen.java
public class Komponen extends Elektronik {
    private String kode;
    private String tipe;

    public Komponen(String merk, int tahunProduksi, double harga, String kode, String tipe) {
        super(merk, tahunProduksi, harga);
        this.kode = kode;
        this.tipe = tipe;
    }

    public String getKode() { return kode; }
    public void setKode(String kode) { this.kode = kode; }

    public String getTipe() { return tipe; }
    public void setTipe(String tipe) { this.tipe = tipe; }

    @Override
    public void displayInfo() {
        super.displayInfo();
        System.out.println("Kode: " + kode);
        System.out.println("Tipe: " + tipe);
    }
}
